packages/client/src/utils/id-token.test.ts

```diff
@@ -118,12 +118,12 @@ describe('decodeToken', () => {
       .setIssuedAt()
       .setExpirationTime('2h')
       .sign(privateKey);
-    const payload = decodeToken(JWT);
-    expect(payload.sub).toEqual('foz');
+    const idTokenClaims = decodeIdToken(JWT);
+    expect(idTokenClaims.sub).toEqual('foz');
   });
 
   test('throw on invalid JWT string', async () => {
-    expect(() => decodeToken('invalid-JWT')).toThrow();
+    expect(() => decodeIdToken('invalid-JWT')).toThrow();
   });
 
   test('throw ZodError when iss is missing', async () => {
```

Comment:
should be `StructError`?

Reply:
`decodeIdToken` throws `StructError` (imported via SuperStruct) only when SuperStruct `assert` throw `StructError`, otherwise it throws a normal `Error` for now.

```ts
export const decodeIdToken = (token: string): IDToken => {
  const payloadPart = token.split('.')[1];

  if (!payloadPart) {
    throw new Error('invalid token');
  }

  const payloadString = payloadPart.replace(/-/g, '+').replace(/_/g, '/');
  const json = decodeURIComponent(
    escape(Buffer.from(fullfillBase64(payloadString), 'base64').toString())
  );

  try {
    // Using SuperStruct to validate the json type
    const idToken = JSON.parse(json) as IDToken;
    s.assert(idToken, IDTokenSchema);
    return idToken;
  } catch (error: unknown) {
    if (error instanceof s.StructError) {
      throw error;
    }

    throw new Error('invalid token: JSON parse failed');
  }
};
```

Should we throw `StructError` instead of normal `Error` all the time here?

If any string cannot be decoded to ID Token, we seems reasonable to assert that there is a struct error in this string.

Reply:
Here is a normal `Error`.
I will assert its content later

Reply:
You mean `decodeIdToken` should not throw `StructError`?